### PR TITLE
Collect box related functions into a box module

### PR DESCRIPTION
**Pull request by yd8534976**
As #491 , to reduce length of each file and make them have single responsibility, this PR collects some box related functions into a `box.py` module.
I can not decide should we need another module to put tf version box functions into.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature / Optimization (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: lmnet/lmnet/utils/box.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+def iou(boxes, box):
+    """Calculate overlap
+
+    Args:
+        boxes: boxes in the image. shape is [num_boxes, 4 or more(x, y, w, h, ...)]
+        box: a single box in the image. shape is [4 or more (x, y, w, h, ...)]
+    Returns:
+        iou: shape is [num_boxes]
+    """
+    assert boxes.size != 0, "Cannot calculate if ground truth boxes is zero"
```
Comment (tsawada) [inline lmnet/lmnet/utils/box.py]: don't use assert to test if the function is used correctly.
use `if` and raise a `ValueError`

File: lmnet/lmnet/utils/box.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+    return intersection / (union + epsilon)
+
+
+def fill_dummy_boxes(gt_boxes, num_max_boxes):
+    dummy_gt_box = [0, 0, 0, 0, -1]
+    if len(gt_boxes) == 0:
+        gt_boxes = np.array(dummy_gt_box * num_max_boxes)
+        return gt_boxes.reshape([num_max_boxes, 5])
+    elif len(gt_boxes) < num_max_boxes:
```
Comment (tsawada) [inline lmnet/lmnet/utils/box.py]: `if`

File: lmnet/lmnet/utils/box.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    """Crop boxes
+
+    Args:
+        boxes: ground truth boxes in the image. shape is [num_boxes, 5(x, y, w, h, class)]
+        crop_rect: a crop rectangle. shape is [4(x, y, w, h)]
+    Returns:
+        cropped_boxes: shape is [num_boxes, 5(x, y, w, h, class)]
+    """
+    # check crop_rect overlap with boxes.
+    assert not ((crop_rect[0] + crop_rect[2]) < boxes[:, 0]).any()
```
Comment (tsawada) [inline lmnet/lmnet/utils/box.py]: same

File: lmnet/lmnet/data_augmentor.py [hunk 4]
```diff
@@ -22,6 +22,9 @@
 from PIL import Image, ImageEnhance, ImageFilter
 
 from lmnet import data_processor, pre_processor
+from lmnet.utils.box import fill_dummy_boxes
```
Comment (tsawada) [inline lmnet/lmnet/data_augmentor.py]: won't this fit in a single line?

File: lmnet/lmnet/utils/box.py (hunk 1 above)
Comment (yd8534976) [reply lmnet/lmnet/utils/box.py]: It is a legacy code, I changed nothing but place. Need I change this function body itself? 🤔

**Pull request merged**